### PR TITLE
Add Volume Fixture

### LINKED ISSUE #70: Add volume fixture
As a user, I'd like to be able to create a temporary UC Volume for testing. 

I am happy to contribute this feature with guidance as required.

### PR DESCRIPTION
**Pull request by christophergrant**
## Changes
This PR adds a Managed Volume fixture from Unity Catalog, allowing testers to create and use a random volume in the catalog.

### Linked issues
Resolves #70 .

### Tests

- [x] manually tested
- [x] added unit tests
- [x] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: tests/unit/fixtures/test_catalog.py [hunk 1]
```diff
@@ -157,3 +158,23 @@ def test_make_schema() -> None:
         full_name='hive_metastore.abc',
         storage_location='abfss://container1@storage.com',
     )
+
+
+def test_make_volume():
+    ctx, info = call_stateful(make_volume)
+
+    ctx['ws'].volumes.create.assert_called_once()
+    print(info)
```
Comment (nfx) [inline tests/unit/fixtures/test_catalog.py:167]:
```suggestion
```

File: tests/unit/fixtures/test_catalog.py [hunk 2]
```diff
@@ -157,3 +158,23 @@ def test_make_schema() -> None:
         full_name='hive_metastore.abc',
         storage_location='abfss://container1@storage.com',
     )
+
+
+def test_make_volume():
+    ctx, info = call_stateful(make_volume)
+
+    ctx['ws'].volumes.create.assert_called_once()
```
Comment (nfx) [inline tests/unit/fixtures/test_catalog.py:166]: can you assert on the actual arguments? :) then we're good to merge

File: tests/unit/fixtures/test_catalog.py [hunk 3]
```diff
@@ -157,3 +158,23 @@ def test_make_schema() -> None:
...
+    print(info)
+    assert info.catalog_name is not None
+    assert info.schema_name is not None
+    assert info.name is not None
+
+
+def test_make_volume_with_name():
+    ctx, info = call_stateful(make_volume, name='test_volume')
+
+    ctx['ws'].volumes.create.assert_called_once()
```
Comment (nfx) [inline tests/unit/fixtures/test_catalog.py:176]: can you assert on the actual arguments? :) then we're good to merge

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/unit/fixtures/test_catalog.py (hunk 2 above)
Comment (christophergrant) [reply tests/unit/fixtures/test_catalog.py:165]: PTAL after changes

File: tests/unit/fixtures/test_catalog.py (hunk 3 above)
Comment (christophergrant) [reply tests/unit/fixtures/test_catalog.py]: PTAL after changes

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**